### PR TITLE
Chore: Mark dio as stable

### LINKED ISSUE #913: Remove `experimental` annotation of `sentry_dio`
I believe it's fair to remove the `experimental` annotation from `sentry_dio`. It's been multiple months without any reported bugs, even though it has a popularity of 91%.

### PR DESCRIPTION
**Pull request by ueman**
## :scroll: Description
Mark Dio as stable.
@marandaneto what would be the correct changelog for this?

## :bulb: Motivation and Context
Closes #913

## :green_heart: How did you test it?
No code changes

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps
Release 🚀

### CODE REVIEW

File: dio/pubspec.yaml [hunk 1]
```diff
@@ -11,12 +11,11 @@ environment:
 dependencies:
   dio: ^4.0.0
   sentry: ^6.6.1
-  meta: ^1.3.0
 
 dev_dependencies:
+  meta: ^1.3.0
   lints: ^2.0.0
   test: ^1.21.1
-  yaml: ^3.1.0 # needed for version match (code and pubspec)
```
Comment (marandaneto) [inline dio/pubspec.yaml:19]: Should we rather keep and add the `version_test`?

**Comment by marandaneto** [comment]
@ueman changelog can be as simple as `Dio is promoted to GA and not experimental anymore`, something like that.

File: dio/pubspec.yaml (hunk 1 above)
Comment (ueman) [reply dio/pubspec.yaml:19]: Sure

**Review by marandaneto — APPROVED** [review]

**Comment by marandaneto** [comment]
Thanks @ueman

File: dio/pubspec.yaml (hunk 1 above)
Comment (ueman) [reply dio/pubspec.yaml:19]: https://github.com/getsentry/sentry-dart/issues/919

**Pull request merged**